iroha_data_model/src/events.rs

```diff
@@ -229,7 +231,86 @@ pub mod pipeline {
     }
 
     /// The reason for rejecting transaction.
-    pub type RejectionReason = String;
+    #[derive(Debug, Clone, Eq, PartialEq, Serialize, Deserialize, Decode, Encode)]
+    pub enum RejectionReason {
+        /// Genesis account can sign only transactions in the genesis block.
+        GenesisAccountSign,
+        /// Failed to verify signatures
+        VerifySignatures {
+            /// Signature which verification has failed
+            signature: Signature,
+            /// Error which happened during verification
+            reason: String,
+        },
+        /// Failed verify signature condition specified in the account
+        SatisfySignatureCondition {
+            /// Reason why signature condition failed
+            reason: String,
+        },
+        /// Failed execute instruction
+        ExecuteInstruction {
+            /// Instruction which execution failed
+            instruction: Instruction,
+            /// Error which happened during execution
+            reason: String,
+        },
+
+        /// Block was rejected during consensus.
+        //TODO: store rejection reasons for blocks?
+        ConsensusBlockRejection,
```

Comment:
All the other enum variants are related to transaction, please separate enums for transaction rejection reason and block rejection reason.
```rust
enum RejectionReason{
    Block(BlockRejectionReason),
    Transaction(TransactionRejectionReason)
}
```